test/test_euler.py

```diff
@@ -410,6 +446,8 @@ def test_facial_flux(actx_factory, order, dim):
         bf_split = split_conserved(dim, boundary_flux)
         assert fnorm(bf_split.mass) < tolerance
         assert fnorm(bf_split.energy) < tolerance
+        if nspecies > 1:
+            assert fnorm(bf_split.massfractions) < tolerance
```

Comment:
We should make it so that this succeeds (i.e. the norm returns 0) for an empty object array, to avoid the branch.